python/cudf/cudf/core/reshape.py

```diff
@@ -238,10 +255,15 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
         result_columns = objs[0].columns
         for o in objs[1:]:
             result_columns = result_columns.append(o.columns)
-
-        df.columns = result_columns.unique()
         if ignore_index:
-            df.index = cudf.RangeIndex(len(objs[0]))
+            df.columns = pd.RangeIndex(len(result_columns.unique()))
+        else:
+            df.columns = result_columns.unique()
+        if ignore_index and not join == "inner":
+            indexes = []
+            for obj in objs:
+                indexes.append(len(obj.index))
+            df.index = cudf.RangeIndex(max(indexes))
```

Comment:
Since we're just looking for the longest object here we can simplify this a bit:
```suggestion
            df.index = cudf.RangeIndex(max(len(obj) for obj in objs)
```

Reply:
Yes! Thank you!